### PR TITLE
MDEV-29101 updated help options for `mariabackup`

### PR DESCRIPTION
**Pull request by gulshanpr**
- [x] *The Jira issue number for this PR is: MDEV-29101*

## Description
This PR is based on #2808, closed due to making blank commit sincere apologies for that.
This patch is for updating the client side `mariabackup --help` options, some of them are deprecated from version 10.5 .

## How can this PR be tested?
This PR is tested manually on Windows using `server\bld\mysql-test>perl mysql-test-run.pl mariabackup.deprecated_options`.
by executing `mariabackup --help` in test.

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch.*
- [x] *This is a bug fix and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.0/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [x] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

File: extra/mariabackup/xtrabackup.cc [hunk 1]
```diff
@@ -1194,23 +1194,19 @@ struct my_option xb_client_options[]= {
...
-     "--compress is used without an argument.",
+     "This option was deprecated starting with MariaDB 10.1.31 and 10.2.13 "
+	 "as it relies on the no longer maintained QuickLZ library.",
      (G_PTR *) &xtrabackup_compress_alg, (G_PTR *) &xtrabackup_compress_alg, 0,
      GET_STR, OPT_ARG, 0, 0, 0, 0, 0, 0},
 
     {"compress-threads", OPT_XTRA_COMPRESS_THREADS,
-     "Number of threads for parallel data compression. The default value is "
-     "1.",
+     "This option was deprecated with --compress ",
```
Comment (bnestere) [inline extra/mariabackup/xtrabackup.cc:1203]: Hi Gulshan!

Thanks for the patch! I have a couple quick notes:

1.  I don't think we want to get rid of the existing description text; I'd imagine we should just extend it with the deprecation message so we don't lose anything.

 2. I think the end part mentioning "with --compress" is a little ambiguous too. I'd suggest omitting it altogether, or replacing it with the "as it relies on the no longer maintained QuickLZ library." to be explicit (and the same goes for `--chunk-size` and `decompress`).

Comment (gulshanpr) [reply extra/mariabackup/xtrabackup.cc:1203]: @bnestere thanks for you suggestions, i will make the changes.

**Review by grooverdan — APPROVED** [review]

**Comment by bnestere** [comment]
@grooverdan If the 'effective' deprecation date of this is 10.1.31, would it make sense for this patch to go into 10.4, rather than 10.5? (though note the JIRA says 10.5)

**Comment by grooverdan** [comment]
> @grooverdan If the 'effective' deprecation date of this is 10.1.31, would it make sense for this patch to go into 10.4, rather than 10.5? (though note the JIRA says 10.5)

Good idea. Rebased now.

**Review by bnestere — APPROVED** [review]
Looks good. Thanks @gulshanpr!

**Pull request merged**